neuraxle/base.py

```diff
@@ -442,6 +466,57 @@ def empty(self):
         """
         return len(self) == 0
 
+    def load(self, path: str) -> 'BaseStep':
+        """
+        Load full dump at the given path.
+
+        :param path: pipeline step path
+        :type path: str
+        :return: loaded step
+        :rtype: BaseStep
+
+        .. seealso::
+            :class:`FullDumpLoader`,
+            :class:`Identity`
+        """
+        context_for_loading = self
+        context_for_loading = context_for_loading.push(Identity(name=path))
+
+        if os.sep in path:
+            context_for_loading = context_for_loading.to_identity()
+            path = path.split(os.sep)[-1]
```

Comment:
The line here seems error-prone: 
`path = path.split(os.sep)[-1]`

1. why use path and not `self.get_names()[-1]` instead?
2. why just the last one?
3. you could extract private method from this if to verbalize what it does and which would answer my 2 questions above.